pkg/asset/machines/master.go

```diff
@@ -544,7 +542,7 @@ func (m *Master) Machines() ([]machineapi.Machine, error) {
 	libvirtapi.AddToScheme(scheme)
 	openstackapi.AddToScheme(scheme)
 	ovirtproviderapi.AddToScheme(scheme)
-	vsphereapi.AddToScheme(scheme)
+	machineapi.AddToScheme(scheme)
```

Comment:
See https://github.com/openshift/installer/pull/5350#pullrequestreview-806725545 for why this won't work.

Reply:
@staebler Since there are multiple PRs attempting to solve this issue, what do you recommend?

Reply:
@kirankt is working on it. I recommend waiting for his fix. Is this something that is blocking this PR?

Reply:
@staebler The issue here is that the cluster-api-provider-alibaba requires a newer version of the machine-api-operator. The vsphere provider requires an older version. Those conflict based upon the machine libraries moving to openshift/api. There might be some go.mod pinning to work around this but I wanted to provide the correct fix by cleaning up the installer dependencies and moving the requirement to openshift/api. I will wait to see what @kirankt comes up with unless you have a suggestion.

Reply:
I don't have a good solution. The recent vendoring brought back MAO probably because of cap-alibaba deps.

Reply:
@kirankt The issue here isn't cap-Alibaba. The dependency for mao in cap-Alibaba is recent. The issue is we need to clean up the vsphere libraries. The vsphere and openstack caps depend on the older version. I created merged PRs in both of those repositories to update the mao dependency.

Reply:
@kirankt I checked out your PR. It appears that you are adding the scheme and addressing @staebler's feedback. I'll await  your changes and will rebase when yours completes. Thanks